og.module

```diff
@@ -44,6 +45,62 @@ function og_entity_insert(EntityInterface $entity) {
 }
 
 /**
+ * Implements hook_og_group_created().
+ */
+function og_og_group_created($entity_type, $bundle) {
+  $fields = OgGroupAudienceHelper::getAllGroupAudienceFields('user', 'user');
+
+  foreach ($fields as $field) {
+
+    if ($field->getFieldStorageDefinition()->getSetting('target_type') == $entity_type) {
+
+      if (!$field->getSetting('handler_settings')['target_bundles']) {
+        return;
+      }
+
+      if (in_array($bundle, $field->getSetting('handler_settings')['target_bundles'])) {
+        return;
+      }
+    }
+  }
+
+  // If we reached here, it means we need to create a field.
+  // Pick an unused name.
```

Comment:
Hmm yes you're right. I was thinking from a data storage perspective, but we also need to be able to manage user memberships from the user edit form. A field is the only good practical way to do this. The alternative would be to alter the user edit form to inject our own form fields, but that's silly - we would lose all the benefits from the field API. So yeah we do need this field.

I see that we are already accounting for our custom storage in by returning an empty array in `OgMembershipReferenceItem::schema()`.